### PR TITLE
Use @Configuration in detekt-rules-style

### PR DESCRIPTION
**Pull request by marschwar**
This belongs to #3670 and replaces all configuration kdoc tags in detekt-rules-style.

### CODE REVIEW

File: detekt-core/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -681,6 +681,7 @@ style:
   ThrowsCount:
     active: true
     max: 2
+    excludeGuardClauses: false
```
Comment (marschwar) [inline detekt-core/src/main/resources/default-detekt-config.yml:684]: This seems to have been an undocumented feature

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumber.kt [hunk 2]
```diff
@@ -101,21 +82,40 @@ class MagicNumber(config: Config = Config.empty) : Rule(config) {
...
+    private val ignoreConstantDeclaration: Boolean by config(true)
+
+    @Configuration("whether magic numbers in companion object declarations should be ignored")
+    private val ignoreCompanionObjectPropertyDeclaration: Boolean by config(true)
+
+    @Configuration("whether magic numbers in annotations should be ignored")
+    private val ignoreAnnotation: Boolean by config(false)
+
+    @Configuration("whether magic numbers in named arguments should be ignored")
+    private val ignoreNamedArgument: Boolean by config(true)
```
Comment (marschwar) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumber.kt:109]: This previously was false by default but the documentation had true as the default. I opted to change it as the generated default config yaml was generated from the documentation.

**Review by schalkms — APPROVED** [review]

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt [hunk 3]
```diff
@@ -50,8 +45,21 @@ class ForbiddenMethodCall(config: Config = Config.empty) : Rule(config) {
...
+            "with this concrete signature."
+    )
+    private val methods: List<String> by config(
+        listOf(
+            "kotlin.io.println",
+            "kotlin.io.print"
+        )
+    )
+
+    private val forbiddenMethods = methods.map { extractMethodNameAndParams(it) }
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt:62]: This one is to use a better name in the rule, right?

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLength.kt [hunk 4]
```diff
@@ -92,10 +93,6 @@ class MaxLineLength(config: Config = Config.empty) : Rule(config) {
     }
 
     companion object {
-        const val MAX_LINE_LENGTH = "maxLineLength"
         const val DEFAULT_IDEA_LINE_LENGTH = 120
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxLineLength.kt:96]: Can't this be private? or removed?

**Review by BraisGabin — APPROVED** [review]

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt (hunk 3 above)
Comment (marschwar) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt:62]: To be honest, primarily I did not like the type of the field in combination with its name. 
```kotlin
    private val methods: List<Pair<String, List<String>?>> by config...
```
I can change it for consistency.

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstant.kt [hunk 5]
```diff
@@ -51,18 +46,24 @@ class FunctionOnlyReturningConstant(config: Config = Config.empty) : Rule(config
...
-    )
-        .map { it.removePrefix("*").removeSuffix("*") }
+    @Configuration("if overriden functions should be ignored")
+    private val ignoreOverridableFunction: Boolean by config(true)
+
+    @Configuration("if actual functions should be ignored")
+    private val ignoreActualFunction: Boolean by config(true)
+
+    @Configuration("excluded functions")
+    private val excludedFunctions: SplitPattern by config("describeContents") { SplitPattern(it) }
```
Comment (marschwar) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstant.kt:56]: Why does this default to "describeContents"? Is this a misconfiguration?
Should we improve the documentation while we are at it?

Comment (marschwar) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/FunctionOnlyReturningConstant.kt:56]: I looked at the original commit and this was clearly not intended to have `'describeContents`' as the default.

**Pull request merged**